doc/design/ib-vf-configuration.md

```diff
@@ -0,0 +1,115 @@
+---
+title: IB VF GUID Configuration
+authors:
+  - almaslennikov
+reviewers:
+  - ykulazhenkov
+creation-date: 11-03-2024
+last-updated: 13-03-2024
+---
+
+# IB VF GUID Configuration
+
+## Summary
+Allow SR-IOV Network Operator to use a static configuration file from the host filesystem to assign GUIDs to IB VFs
+
+## Motivation
+We have customers using the SR-IOV operator to create IB VFs, and they need a way to automate GUID assignment,
+so that IB VFs are automatically bound to the required PKeys and no additional manual configuration is needed.
+In this use case, the GUID configuration is static and known in advance.
+Now the GUIDs are assigned by the sriov-network-config-daemon randomly.
+
+### Use Cases
+
+### Goals
+
+* IB GUID configuration can be read from a static json file on the host
+* IB GUID configuration is static and created in advance
+* Static config file is available at the same path on every host
+
+### Non-Goals
+
+* Dynamic GUID allocation is out of scope of this proposal
+
+## Proposal
+
+Mount a configuration file as a part of `/host` hostPath in the config daemon and read it to retrieve the GUID configuration.
+
+### Workflow Description
+
+In the [`pkg/host/internal/sriov/sriov.go:configSriovVFDevices`](https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/82a6d6fdce71bd88a0d9368fb1750488e9a8e4e2/pkg/host/internal/sriov/sriov.go#L458) read from the static IB config file
+and assign the GUIDs according to the configuration. Each PF is described by either PCI address
+or the PF GUID and has either a list or a range of VF GUIDs.
+
+1. A script creating the config file is deployed to the host and creates a static GUID config file
+2. SR-IOV network operator reads the file and assign GUIDs when IB VFs are created
+3. Users employ [PKey selector](https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pull/517) in sriov-network-device-plugin to create PKey-specific resource pools
```

Comment:
how this will get populated to the device plugin via the sriov-network-operator?

Reply:
Made a note on that in the doc. Essentially, once the GUID is assigned, the corresponding PKey is stored in the sysfs. The device plugin takes it from there to populate resouce pools with PKey selector enabled